### DISCUSSION TITLE
Guide: system-wide audio EQ using AU Lab

### DISCUSSION THREAD

**Discussion by clanger9** [ExistentialAudio/BlackHole#474] [General]
I made a guide for doing system-wide audio EQ using AU Lab. It includes the necessary steps to stop it clicking.
All errors and omissions are mine. I hope this is useful for someone. Feel free to reuse/republish.

### Download
1.	BlackHole 2ch (free, email required): https://existential.audio/blackhole/
2.	Additional Tools for Xcode 13 (includes AU Lab v2.3, free Apple developer registration required): https://download.developer.apple.com/Developer_Tools/Additional_Tools_for_Xcode_13/Additional_Tools_for_Xcode_13.dmg

### Install
Install BlackHole
Copy AU Lab app to Applications
 
### Configure 
From the macOS menu bar, select **BlackHole 2ch** as your speaker 
![image](https://user-images.githubusercontent.com/13693538/137122197-258a82b3-6c0c-4489-912f-ad85dad55ac7.png)
 
Open _Audio Midi Setup_ app
Click **+ Create Aggregate Device**
Under **Use**, check both **BlackHole 2ch** and your speaker device
Under **Drift Correction**, check both **BlackHole 2ch** and your speaker device <- this is important to stop it clicking & crackling after a few minutes
Click on **Aggregate Device** and rename it to "Desktop Audio Correction"
![image](https://user-images.githubusercontent.com/13693538/137120877-d21b32df-3bd7-40c8-bc6c-e5597765f73d.png)
 
Click on your speaker device name and set channel volume of your speakers to **-10dB**
![image](https://user-images.githubusercontent.com/13693538/137120909-481f5719-3c0c-4f35-89b6-65d1ae15192b.png)
 
Open _AU Lab_ app
 
**AU Lab -> Preferences -> Devices**
Select **Desktop Audio Correction**
Set **Frames** to 1024
Under Expert Settings, set **CPU Usage** to 10%
![image](https://user-images.githubusercontent.com/13693538/137120927-2a54bafd-10b9-44a0-ab28-f91bb6425a8b.png)

**File -> New**
Select **Stereo In/Stereo Out** and **Desktop Audio Correction** as both Input and Output Device
Click **Create Document**
![image](https://user-images.githubusercontent.com/13693538/137120954-de00acca-c8c2-4d62-bcbb-ad17570f6312.png)

Add EQ & room corrections under Output 1 - Effects .
![image](https://user-images.githubusercontent.com/13693538/137120971-9bd0c2da-7f0f-4884-81f7-a5f16c13d203.png)

- For the usual graphic equaliser, select "AUGraphicEQ"
- For room corrections, "AUNBandEQ" is probably the plugin you want…

![image](https://user-images.githubusercontent.com/13693538/137120988-3cb08da5-ccda-4b52-9861-61e48d44f53b.png)

**File -> Save as** "Desktop audio correction"
 
**AU Lab -> Preferences -> Document**
Check **Open a specific document:**, select the file you saved above
Check **Use Preferred Audio Device…** and select **Desktop Audio Correction**
![image](https://user-images.githubusercontent.com/13693538/137122330-0713ca96-5976-4fd9-aa06-21b6df50c6e8.png)

**Window -> Minimise**
Right-click on AU Lab icon in the dock, check **Options -> Open at Login**
 
In **System Preferences -> Users & Groups -> Login Options**, look for the AU Lab line and check Hide 
![image](https://user-images.githubusercontent.com/13693538/137121028-41f2f3d4-a499-4d7a-85aa-e6927b2e699f.png)
 
If you don't get any sound via AU Lab, it may be that it hasn't got permission to receive audio. Check under **System Preferences -> Security & Privacy -> Microphone**

**Comment by porg** [comment]
Hi @clanger9, thanks for providing this guide!
- Before going deeper and trying this out: Is it possible to **apply this equalizer (EQ) only on one speaker** (e.g. left speaker only) ?
  - My left MacBookPro speaker starts to fail slowly (resonates at certain frequency range)
  - As a temporary measure (until repair/replacement) I 'd like to cut off the annoying frequencies, but only on the left channel.

**Reply by clanger9** [reply]
Hi @pog I can't see an obvious way to do this on a single channel.
Since writing this guide I have switched to using Hosting AU (rather than AU Lab), which seems to be more reliable.

**Reply by porg** [reply]
- Thanks for the info that one channel EQ is not possible in AU Lab.
- Would it be in Hosting AU?

**Reply by clanger9** [reply]
Hosting AU seems to have a similar limitation. It _might_ be possible to create an aggregate device in Audio MIDI Setup to route the EQ to one channel only. Just a thought...

**Comment by FrankSidebottom1** [comment]
![Screenshot 2025-06-03 at 19 36 21](https://github.com/user-attachments/assets/eca5fc5b-7531-4c7b-916b-b0521cb62c7d)
The meters show audio but I cannot hear anything :(

**Comment by devinroth** [comment]
BlackHole doesn’t send audio to your speakers. You need to do that yourself either with a daw or multi output.

Devin

**Reply by FrankSidebottom1** [reply]
So if I generate a multi output device in Audio MIDI Setup and set 2 CH BlackHole + UMC4044 (which is my audio interface)  as Output should this work?

**Comment by devinroth** [comment]
Yeah depending on what you’re doing.

Devin

**Reply by FrankSidebottom1** [reply]
I want to apply an EQ to my systems audio. 
The signal flow would be Computer Audio > EQ > Audio Interface (Speakers)
Before I could do it with soundflower and AU Lab but soundflower has no support for my mac mini m2 so I'm taking my chances with Blackhole 2CH but no luck

**Reply by paulzaptec** [reply]
The above instructions should work.
If you want, you can use [Hosting AU](https://ju-x.com/hostingau.html) instead of Apple's AU Lab. I find this works better; I'm using it to do system-wide EQ on my MacBook Pro M1.

**Comment by Headlands7** [comment]
Thank you for this! I just can't get it working, have been through every step multiple times. Is this not working on Sonoma, perhaps?

**Comment by devinroth** [comment]
The usually culprit is mic permissions.

Devin

**Reply by Headlands7** [reply]
Thank you, I did that for everything (deleted all mic permissions on my mac, then enabled them one-by-one) and still no luck...no audio is getting into AU Lab. About to give up and pay the $45 to get whatever that app is that does it all for you. Any other suggestions to try before I do that?

**Reply by pseigo** [reply]
Take a look [below](https://github.com/ExistentialAudio/BlackHole/discussions/474#discussioncomment-13508918), @Headlands7

**Comment by devinroth** [comment]
You can try a DAW instead if AU labs isn’t working.

Devin

**Comment by pseigo** [comment]
Working for me so far on macOS Sonoma 14.7.

1. Follow this guide. (https://github.com/ExistentialAudio/BlackHole/discussions/474)
2. Create a "Multi-Output Device" in _Audio MIDI Setup_: https://github.com/ExistentialAudio/BlackHole/wiki/Multi-Output-Device -- but set BlackHole as your output device.
3. If _AU Lab_ doesn't request Microphone permissions, follow the steps summarized [here](https://github.com/ExistentialAudio/BlackHole/issues/4#issuecomment-2984033273) under issue #4.

p.s., I get ParametricEQ profiles for my headphones (that I type into a 10-band AUNBandEQ effect) from [AutoEq](https://github.com/jaakkopasanen/AutoEq/tree/master/results#readme) :)
